### PR TITLE
Added yaml factory for custom roles and bindings

### PR DESCRIPTION
**Pull request by AlexeyS-CIO**
Added yaml factory:
custom roles,
iam_bindings,
iam_bindings_additive
---
**Checklist**

I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

**Review by ludoo — CHANGES_REQUESTED** [review]
Ach, here are my comments, but then I did not actually look at the code as my objections are more of a fundamental nature.

File: modules/project/iam.tf [hunk 1]
```diff
@@ -33,15 +33,70 @@ locals {
...
+    flatten([for f in try(fileset(var.factories_config.bindings, "*.yaml"), []) :
+      [for binding_name, data in yamldecode(file("${var.factories_config.bindings}/${f}")) : {
+        binding_name = binding_name
+        data = {
+          members = data.members
+          role    = try(google_project_iam_custom_role.roles[data.role], data.role)
+          condition = {
+            title       = try(data.condition.title, "")
+            expression  = try(data.condition.expression, "")
+            description = try(data.condition.description, "")
```
[the remark this answers is not in the document]
Comment (AlexeyS-CIO) [reply modules/project/iam.tf]: done

**Comment by AlexeyS-CIO** [comment]
Fixed, issues.
Moved factory to separated file.
If you have any questions about uses cases, we can do quick meet=)

**Comment by ludoo** [comment]
Alexey, I'm not sure this fits in with the project module: if you need to manage complex roles and IAM it a) probably means the setup could be optimized (folders, etc.), and b) you manage a sizeable number of projects and should use the project factory, which already allows you to do this.

**Comment by ludoo** [comment]
Alexey, I'm closing this but we added a custom roles factory for a separate change we're working on, you will see it merged  soon.

**Pull request closed without merging**